pkg/segment/set.go

```diff
@@ -399,12 +415,21 @@ func (s *Set) Remove(seg Iterator) GapIterator {
 		// overlap.
 		seg.SetRangeUnchecked(victim.Range())
 		seg.SetValue(victim.Value())
+		// Need to update the nextAdjacentNode's maxGap because the gap in between
+		// must have been modified by update seg.Range() to victim.Range().
+		nextAdjacentNode := seg.NextSegment().node
+		nextAdjacentNode.updateMaxGapLeaf()
+		// At this moment the maxGap of the set may not be well maintained because
+		// it's in a state where two segments victim and seg has the same range which
+		// violates the B-tree requirement. It will be fixed after the remove of victim.
```

Comment:
This comment is redundant with the one above that already notes that `seg` and `victim` overlap.